src/spikeinterface/preprocessing/common_reference.py

```diff
@@ -1,25 +1,47 @@
+from __future__ import annotations
 import numpy as np
+from typing import Optional, Literal
 
 from spikeinterface.core.core_tools import define_function_from_class
 
 from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
 from ..core import get_closest_channels
+from spikeinterface.core.baserecording import BaseRecording
 
 from .filter import fix_dtype
 
 
 class CommonReferenceRecording(BasePreprocessor):
     """
-    Re-references the recording extractor traces.
+    Re-references the recording extractor traces. That is, the value of the traces are
+    shifted so the there is a new zero (reference).
+
+    The new reference can be estimated either by using common median reference (CMR) or
+    common average reference (CAR).
```

Comment:
I was thinking that CMR and CAR are the methods used to estimate the new reference. As in, I estimate the error using gradient descent (where gradient descent is the method). Would I stll need the article 'a'?

But also, now that I think about this, this is not an estimation, this is the exact value. See Alessio comments for why I was wrong.

Reply:
This is a bit tricky. Descent is actually a noun whereas descend is the verb. So when you perform gradient descent you are descending the gradient. In this case you would use the article for emphasis, I perform a gradient descent ( I am talking about one step in a series of steps). Reference itself is a verb (to reference something), so the default is in general to treat it like a normal verb where we either use the noun form (a reference) or a gerund which can be used as a noun (in this case referencing). But either way you convert this to a noun version. 

to be fair this is an area of argument in English which as a language has a strong vs weak system. For example neologisms should follow a weak system rather than a strong system. So the new word 'to gaslight' should have a past tense of gaslighted, but because light has an irregular past tense of lit, some people would prefer to have the past tense gaslit. 

But above all, if this is a super domain specific terminology (like you know for sure the field uses CAR/CMR without the article) then you would follow the domain even if it is wrong by standard grammar. I understand CMR as a concept, but I won't pretend to be an expert in the field. So I was correcting that based on standard grammar convention (where my gut tells me to say I perform gradient descent, but also tells me to say I used a common median reference to calculate the new reference value).

Reply:
Thanks for this explanation. I think I trust your gut in this one as we want to make the reading sound idiomatic if possible.